Full Pydantic V2 support & latest MTENN (0.7.8) support

kaminow opened the pull request:
`mtenn` is updating to allow separate treatment of the complex, protein, and ligand, so need to update here for compatibility.

kaminow committed:
Pass export_input_data and export_exp_data args around correctly.

kaminow committed:
Split split_dict validator into one for initializing and one for checking.

kaminow committed:
Add missing args to ds_config.

kaminow committed:
Rework kvp splitting function to allow : in the values (needed for e3nn irreps).

kaminow committed:
Add eplicit optional to Field annotations.

kaminow committed:
Update model building CLI tests.

kaminow committed:
Remove visnet tests for now.

kaminow committed:
__fields__ -> model_fields

kaminow committed:
json -> model_dump_json

kaminow committed:
Need to validate defaults for some Fields.

kaminow committed:
Need to add general_ds_args decorator.

kaminow committed:
Need to add struct_ds_args decorator.

kaminow committed:
Update build-and-train CLI tests.

kaminow committed:
Capitalize split names automaticall to make plots prettier.

kaminow committed:
Filter split datasets for structures with invalid smiles.

kaminow committed:
Update jitter CLI testing.

kaminow committed:
Only keep floats for target_val.

kaminow committed:
Skip tests that won't work yet.

kaminow committed:
Update json roundtrip test.

kaminow committed:
json -> model_dump_json

kaminow committed:
Make sure loss configs are in the same order.

kaminow committed:
Remove unused import.

kaminow committed:
Add option for date-based filtering.

mikemhenry committed:
Merge branch 'main' into pydantic_2_separate_envs

mikemhenry committed:
bump ci

apayne97 committed:
test running with multiple env files

apayne97 committed:
move openeye testing

apayne97 committed:
use different micromamba command

apayne97 committed:
fix missing asapdiscovery-modeling install

kaminow committed:
Add verbose option for printing info for each pred_tracker file.

kaminow committed:
Add handling for trying to parse Nones in pred trackers.

kaminow committed:
dict -> model_dump

ijpulidos committed:
Environment file for cli ci ubuntu

ijpulidos committed:
Environment file for cli ci macos

Commit:
Update ML workflow file to include MacOS and latest micromamba version.

apayne97 committed:
add workflows

ijpulidos committed:
workflow file for CLI CI

Commit:
Rename macOS conda-envs directory.

apayne97 committed:
add workflows

Commit:
Merge branch 'pydantic_2_separate_envs' of https://github.com/choderalab/asapdiscovery into pydantic_2_separate_envs

apayne97 committed:
add workflows

apayne97 committed:
Merge remote-tracking branch 'origin/pydantic_2_separate_envs' into pydantic_2_separate_envs

Commit:
fix workflows for simulation and spectrum

Commit:
Merge branch 'pydantic_2_separate_envs' of github.com:choderalab/asapdiscovery into pydantic_2_separate_envs

Commit:
Remove MacOS from ML workflow.

apayne97 committed:
add workflows

apayne97 committed:
Merge remote-tracking branch 'origin/pydantic_2_separate_envs' into pydantic_2_separate_envs

apayne97 committed:
install environments for other packages

Commit:
Merge branch 'pydantic_2_separate_envs' into add-split-model

kaminow committed:
Fix pose_predictions format.

kaminow committed:
Fix pose_predictions format.

apayne97 committed:
Separate environment files (#9)

* add initial ubuntu and mac latest environments and update data.yaml github workflow to point to it

* add a python script to find imports in files

* add plip dependency

* temporarily fix failing openfe warning catching

* update detect_imports script and add modeling yaml files

* update data and modeling workflows

* update docking CI and add docking yaml

* add dataviz to docking dependencies

* add simulation yamls

* add simulation yamls

* add modeling to simulation dependencies

* add modeling to simulation dependencies

* add macos to data workflow

* update workflows

* remove unnecessary packages from modeling workflow

* add simulation workflow package

* update simulation deps

* add ml env and update env names

* add torchdata to docking deps

* add openmmtools to deps

* add ml to simulation deps

* remove multiprocessing from ML deps

* remove multiprocessing from simulation deps

* update ml envs

* make mtenn simulation dep

* add modeling to ml deps

* add semver to docking deps

* move cli_args to have lazy imports and add spectrum env files

* also run on the pydantic_2 branch

* bump ci

* test running with multiple env files

* move openeye testing

* use different micromamba command

* fix missing asapdiscovery-modeling install

* Update ML workflow file to include MacOS and latest micromamba version.

* add workflows

* Rename macOS conda-envs directory.

* add workflows

* Environment file for cli ci ubuntu

* Environment file for cli ci macos

* workflow file for CLI CI

* add workflows

* fix workflows for simulation and spectrum

* Remove MacOS from ML workflow.

* add workflows

* install environments for other packages

* install environments for other packages

* adding data missing dependencies

* environment file for alchemy ubuntu-latest

* environment file for alchemy macos-latest

* workflow file update for alchemy

* macos uncapitalized in workflows

* Add pydantic-settings.

* add dataviz env files

* Add mamba installing dependencies in ML workflow file.

---------

Co-authored-by: Mike Henry <11765982+mikemhenry@users.noreply.github.com>
Co-authored-by: kaminow <kaminow@cshl.edu>
Co-authored-by: Iván Pulido <2949729+ijpulidos@users.noreply.github.com>
Co-authored-by: Maria Castellanos <castelm3@mskcc.org>
Co-authored-by: kaminow <benjamin.kaminow@choderalab.org>

apayne97 committed:
make a real workflows env file

apayne97 committed:
remove workflow deps yaml

apayne97 committed:
don't run workflows CI on macos

apayne97 committed:
use bioconda for logomaker

apayne97 committed:
Merge branch 'fix_workflows_ci' into move_openeye_functions_to_data

apayne97 committed:
fix dataviz workflow

apayne97 committed:
fix dataviz workflow

apayne97 committed:
update dataviz workflow install

apayne97 committed:
rename dataviz env

apayne97 committed:
fix dataviz env

apayne97 committed:
remove modeling as a dep for data

apayne97 committed:
move trim_small_chains to backend

apayne97 committed:
move trim_small_chains to backend

apayne97 committed:
adapt to str enum update

apayne97 committed:
fix .schema import

apayne97 committed:
fix .schema import

apayne97 committed:
remove macos-ci workflow

apayne97 committed:
Merge pull request #29 from choderalab/remove_old_macos_ci

remove macos-ci workflow

apayne97 committed:
fix from .backend import

apayne97 committed:
Merge remote-tracking branch 'origin/main' into pydantic_2

# Conflicts:
#	.github/workflows/macos-ci.yaml

apayne97 committed:
move prepped structure and docking stuff to modeling and docking envs

apayne97 committed:
Merge pull request #25 from choderalab/fix_workflows_ci

Make workflows CI pass

apayne97 committed:
move plip to dataviz

apayne97 committed:
fix PreppedComplex use in conftests

apayne97 committed:
most postera api to workflows

apayne97 committed:
most postera api to workflows

apayne97 committed:
fix imports

apayne97 committed:
fix imports

apayne97 committed:
test fintscore in dataviz

apayne97 committed:
remove other subpackages for data ci workflow

apayne97 committed:
Merge branch 'pydantic_2' into move_openeye_functions_to_data

apayne97 committed:
fix prep tests

apayne97 committed:
fix prep tests

apayne97 committed:
fix imports

apayne97 committed:
fix imports

apayne97 committed:
fix imports

apayne97 committed:
fix imports

apayne97 committed:
fix imports

apayne97 committed:
Merge pull request #28 from choderalab/move_openeye_functions_to_data

Make data not dependent on any other subpackages

kaminow committed:
Merge from pydantic_2 branch.

kaminow committed:
Update ML env.

kaminow committed:
Fix env files I messed up.

kaminow committed:
Fix env name for installing other drugforge pacakge deps.

kaminow committed:
Remove cudatoolkit from macos simulation.yaml.

kaminow committed:
Remove unittest dep.

kaminow committed:
Make Fields explicitly optional.

kaminow committed:
Fix env name

kaminow committed:
Fix validator for pose_predictions.

kaminow commented:
ok @ijpulidos should be ready for a review whenever you get a chance!

kaminow committed:
Add option for random iterating in Dataset class.

kaminow committed:
Handle random iter field in build method.

kaminow committed:
Add random_iter in DockedDataset builder functions.

kaminow committed:
Add test for random iter.

kaminow committed:
Add CLI args and handling for dataset random_iter option.

kaminow committed:
Add random_iter to the rest of the dataset classes.

kaminow committed:
Add CLI test for random iter.

kaminow committed:
Merge pull request #42 from choderalab/add-data-shuffling

Add option to randomly shuffle dataset

kaminow committed:
Add handling in Trainer for CLI args that are passed along with a cache file.

kaminow committed:
Fix assigning random_iter so that it works for ds objects build before it was introduced.

apayne97 committed:
remove cudatoolkit dep from simulation.yaml (#39)

mikemhenry committed:
switch to using the new buckets (#41)

* switch to using the new buckets

* missed a few urls

---------

Co-authored-by: Alexander Matthew Payne <31800806+apayne97@users.noreply.github.com>

mariacm12 committed:
Spectrum struct changes (#10)

* add rmsd-coloring

* Fix chain visualization

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix pre-commit-ci

* Add the option to align struct directory and pdb file

* Needed changes on seq_alignment

* Implement fitness-alignment to align wrt fasta

* Add more options to struct-alignment

* changes to fitness-alignment

* Additions to align_seq_match

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* precommit ci fixes

* fix seq-align test

* struct-alignment tests

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add structure alignment tests

* Fix struct-alignment cfold test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* bump CI

* bump CI

* Added a check for blast.csv file

* Bump CI

* TODO to fix test here with "hidden" error

* Checking subprocess return code, print error

* added mafft dependancy

* add mafft to ubuntu yaml

* remove macOS skips from tests

---------

Co-authored-by: Castellanos Morales <castelm3@iscb007.mskcc.org>
Co-authored-by: Castellanos Morales <castelm3@iscf019.mskcc.org>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Maria Castellanos <castelm3@mskcc.org>
Co-authored-by: hmacdope <hugomacdermott@gmail.com>
Co-authored-by: Iván Pulido <2949729+ijpulidos@users.noreply.github.com>

apayne97 committed:
Merge branch 'pydantic_2' into add-split-model

ijpulidos committed:
Dropping bespokefit in codebase (#22)

* Dropping bespokefit in codebase

* using pydantic.v1 instead

* trying to fix CI issues by pinning to torchdata 0.8.0 for compatibility with dgl 2.3

* Fixing env path for alchemy CI

* fixing package name for alchemiscale

* fixing package name for alchemiscale on macos

* pinning openfe versions

* Bump CI

* pinning openfe versions for linux

* pinning compatibility to minor versions

* dropping python pins

* dropping python pins for every package

---------

Co-authored-by: Mike Henry <11765982+mikemhenry@users.noreply.github.com>
Co-authored-by: chrisiacovella <chris.iacovella@gmail.com>

apayne97 committed:
Merge branch 'pydantic_2' into add-split-model

apayne97 committed:
update cli-ci (#48)

* update cli-ci

* change order of OE license checks

* add simulation to cli deps

* add dataviz and spectrum to deps

apayne97 committed:
Merge branch 'pydantic_2' into add-split-model

chrisiacovella committed:
fixing pydantic issues in alchemy (#52)

* FIXING ISSUES with pydantic models to handle ScopedKey (including serialization); fixed issues with passing json files to openfe 
---------

Co-authored-by: Alexander Matthew Payne <31800806+apayne97@users.noreply.github.com>

apayne97 committed:
Merge branch 'pydantic_2' into add-split-model

ijpulidos commented:
We made some changes in #60 that should be undo or reviewed when the ML subpkg is fully working with all the others. Just making this note here for reference.

kaminow committed:
Add weights_path in model_config.

kaminow committed:
Migrate ml.inference to pydantic 2.

kaminow committed:
Migrate ml.models to pydantic 2.

kaminow committed:
Missed changes.

kaminow committed:
Add mtenn pins for model specs.

kaminow committed:
Update inference to new mtenn version.

kaminow committed:
Update GAT inference -> LigandOnly inference.

kaminow committed:
Add new ML registry to test files.

kaminow committed:
Handle HttpUrl Fields for urljoin.

Update structure-based model inference tests.

kaminow committed:
Update structure-based model inference tests.

kaminow committed:
Track representation types in model specs.

kaminow committed:
Update test_models hash.

kaminow committed:
Add model selection from representation type.

kaminow committed:
Return input_data format from test_data.

kaminow committed:
Merge from main.

kaminow committed:
Clean up leftover asapdiscovery files.

kaminow committed:
Update ML workflow to use the right conda env.

kaminow committed:
Update alchemy gufe import.

kaminow committed:
Handle label mappings not being passed.

kaminow committed:
Merge branch 'main' into add-split-model

kaminow committed:
Add early stopping that checks for loss below a certain threshold.

kaminow committed:
Start to add analysis tools.

kaminow committed:
Add calc_stats to analysis.

kaminow committed:
Add filtering functions.

kaminow committed:
Add new stat function to calculate MAE that's adjusted for semi-quantitative values.

ijpulidos committed:
Merge branch 'main' into add-split-model

kaminow committed:
Add GL loss class from book.

kaminow committed:
Add PQ loss from book.

kaminow committed:
Add progress_quotient es handling to Trainer.

kaminow committed:
Add config handling for progress quotient es.

kaminow committed:
Missed adding train loss to tracked vals.

kaminow committed:
Need to add to list every epoch.

kaminow committed:
Don't return early.

kaminow committed:
Fix training strip math.

chrisiacovella committed:
resolving pydantic 2 issues in everything but alchemy. (#108)

* updated pydantic v1 models to v2
* updating all Config classes to ConfigDict.  
* I  had to remove the multimethod overloading for dispatch function and just put in logic to switch in the dispatch function, as the multi method decorator does not seem to work with pydantic2
* skip mac os pip install of drugforge ml in cli.yaml
* pinning openmmforcefields to  >=0.15.0 to resolve issues with openmmforcefields system setup on macOS.

---------

Co-authored-by: kaminow <51923685+kaminow@users.noreply.github.com>

kaminow committed:
Merge branch 'main' into add-split-model

kaminow committed:
Handle arbitrary extract epochs.

kaminow committed:
Fix how model_weights are assigned when continuing training.

kaminow committed:
New analysis method to get training progress.

kaminow committed:
Make build_results_dfs a function that can return the dfs if being used not on the CLI.

kaminow committed:
Add filter-date for build_results_dfs.

kaminow committed:
Return None if no pred trackers could be loaded.

kaminow committed:
Fix manual splitting.

kaminow committed:
Use existing machinery for manual splitting when continuing training run.

kaminow committed:
Only need to make wandb tables if we're not continuing a run.

kaminow committed:
Actually need splits for wandb initialization if we're not continuing.

kaminow committed:
Add catches for not loading any pred trackers.

kaminow committed:
Adding statistical analyses.

kaminow committed:
Try running ANOVA on a single thing at a time, but on everything.

kaminow committed:
Pass single column to run on instead of doing everything in one.

kaminow committed:
Keep backup of pred_tracker during training.

kaminow committed:
Check for backup pred_tracker file.

kaminow committed:
Merge branch 'main' into add-split-model

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

kaminow committed:
Remove old comment.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
Rev pydantic2 for alchemy (#132)

* fixing issue in to_rdkit functionality and other minor parts

* Adding env yaml file (all.yaml) that seems to work for creating a single workable environment

* Seems to be an issue with lomap

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Revising code to work with new Gufe that dropped FloatQuantity

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Removing print/debugging that exhausts the file pointer contents

* Need to use raw values instead, because pydantic transforms Quantity to dict

* We force quantities now. Need to review uncertainties.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update macos alchemy env

* Remove reference to `pkg_resources`. Using `importlib`

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Remove reference to `pkg_resources`... 2nd iteration :)

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Using explicit pkg name instead

* Using explicit pkg name instead

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fixing some minor pydantic updates

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Using explicit pkg name instead

* Removing unnecessary debug prints

* Making sure we preserve names when creating oe molecules

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Overriding default settings for rdkit to make serialization work properly

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Overriding default settings for rdkit to make serialization work properly; trying putting it in a different spot.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Overriding default settings for rdkit to make serialization work properly; trying putting it in a different spot.

* Fix some import issues pre-commit found; fixing ml issues.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* fixing precommit complaints

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* xskip on mac os the test_plan_from_names until we can get the issue resolved on GUFE. Also added in check to ensure we are setting AllProps for pickling in rdkit (things will definitely fail if this is not set, even if this is not sufficient on mac os)

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* addressing precommit formatting complaints

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* use lazy imports for docking score test

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* update pin in all.yaml

* update pin in all.yaml

* addressing comments from Alex.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* A few deprecated pydantic functions were still being used.  updating these.

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Iván Pulido <2949729+ijpulidos@users.noreply.github.com>
Co-authored-by: David Dotson <dotsdl@gmail.com>
Co-authored-by: apayne97 <alex.payne@choderalab.org>

chrisiacovella committed:
Merge branch 'main' into add-split-model

# Conflicts:
#	devtools/detect_imports.py
#	drugforge-alchemy/drugforge/alchemy/schema/fec.py
#	drugforge-alchemy/drugforge/alchemy/schema/prep_workflow.py
#	drugforge-data/drugforge/data/readers/structure_dir.py
#	drugforge-data/drugforge/data/schema/complex.py
#	drugforge-data/drugforge/data/schema/target.py
#	drugforge-data/drugforge/data/services/fragalysis/fragalysis_reader.py
#	drugforge-data/drugforge/data/services/postera/manifold_data_validation.py
#	drugforge-data/drugforge/data/testing/test_resources.py
#	drugforge-data/drugforge/data/tests/test_ligand_schema.py
#	drugforge-docking/drugforge/docking/docking.py
#	drugforge-docking/drugforge/docking/fint_scorer.py
#	drugforge-docking/drugforge/docking/meta_scorer.py
#	drugforge-docking/drugforge/docking/ml_scorer.py
#	drugforge-docking/drugforge/docking/schema/pose_generation.py
#	drugforge-docking/drugforge/docking/selectors/mcs_selector.py
#	drugforge-docking/drugforge/docking/selectors/selector.py
#	drugforge-docking/drugforge/docking/tests/test_scorers.py
#	drugforge-ml/drugforge/ml/cli_args.py
#	drugforge-ml/drugforge/ml/inference.py
#	drugforge-ml/drugforge/ml/pretrained_models.py
#	drugforge-ml/drugforge/ml/tests/test_mlops_cli.py
#	drugforge-modeling/drugforge/modeling/protein_prep.py
#	drugforge-modeling/drugforge/modeling/schema.py
#	drugforge-modeling/drugforge/modeling/tests/test_protein_prep.py
#	drugforge-spectrum/drugforge/spectrum/score.py
#	drugforge-spectrum/drugforge/spectrum/tests/test_score.py
#	drugforge-workflows/drugforge/workflows/docking_workflows/cross_docking.py
#	drugforge-workflows/drugforge/workflows/docking_workflows/large_scale_docking.py
#	drugforge-workflows/drugforge/workflows/docking_workflows/ligand_transfer_docking.py
#	drugforge-workflows/drugforge/workflows/docking_workflows/small_scale_docking.py
#	drugforge-workflows/drugforge/workflows/docking_workflows/symexp_crystal_packing.py
#	drugforge-workflows/drugforge/workflows/spectrum_workflows/cli.py
#	drugforge-workflows/drugforge/workflows/spectrum_workflows/score_complex.py

chrisiacovella committed:
removing pkg_resources usage in manifold_data_validation.py

chrisiacovella committed:
Merge branch 'main' into add-split-model

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
fixing pre-commit unused/duplicate imports

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

# Conflicts:
#	drugforge-data/drugforge/data/services/postera/manifold_data_validation.py
#	drugforge-modeling/drugforge/modeling/schema.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
fixing typing issue of conf_tags in ligand.py (needs to be dict)

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

chrisiacovella committed:
Merge branch 'main' into add-split-model

# Conflicts:
#	devtools/conda-envs/macos-latest/spectrum.yaml
#	drugforge-spectrum/drugforge/spectrum/score.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
reformatting score.py

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
updating pins to be consistent to hopefully speed up CI environment solving.

chrisiacovella committed:
to use dask-cuda, it appears we need to use the rapidsai channel for newer versions (they seem to have stopped updating their conda-forge builds)

chrisiacovella committed:
mispelled channel in ml.yaml

chrisiacovella committed:
trying a different strategy for building environment from multiple yaml files for cli

chrisiacovella committed:
trying a different strategy for building environment from multiple yaml files for cli

chrisiacovella committed:
trying a different strategy for building environment from multiple yaml files for workflows and dataviz as it worked for cli

chrisiacovella committed:
need os dependent yaml file args

chrisiacovella committed:
need os dependent yaml file args

chrisiacovella committed:
need os dependent yaml file args

chrisiacovella committed:
need os dependent yaml file args

chrisiacovella committed:
typo in mac os yaml file args

chrisiacovella committed:
forgo to remove some additional conda solving in macos on cli.

chrisiacovella commented:
Note, I had to add in the `rapidsai` channel for dask-cuda.  the condo-forge version is really out of date and it looks like they are distributing on their own channel.   Newer versions of openfe, which are required for pydantic v2, have specific requirements for dask>=2025 (latest on conda-forge was 2023).   

I think we'll just have to update documentation about an extra step to install dask. 


Additionally, I reworked the conda environment building section.  It was taking way too long for workflows and CLI, as we basically would build a minimal environment and then update that environment multiple times for each of the  conda environments for sub packages.   I specified which yaml files we need in the matrix section (so ubuntu and macOS can each have different ones, needed because ml doesn't work on Mac OS )...these are then passed as arguments when we initially build the environment.  So it tries to simultaneously solve, rather than serially.  This also allows better caching.

chrisiacovella committed:
accidentally deleted env: tag in cli.yaml (before we set the license.

apayne97 committed:
fix logomaker error (#141)

apayne97 commented on .github/workflows/ml.yaml:
```diff
@@ -74,11 +75,11 @@ jobs:
+        micromamba install -n drugforge --file devtools/conda-envs/${{ matrix.os }}/docking.yaml
+        micromamba install -n drugforge --file devtools/conda-envs/${{ matrix.os }}/modeling.yaml
+        micromamba install -n drugforge --file devtools/conda-envs/${{ matrix.os }}/alchemy.yaml
+        micromamba install -n drugforge --file devtools/conda-envs/${{ matrix.os }}/cli.yaml
```
do we want to move these to the `yaml_file_args` as we have done in the other workflow yamls?

chrisiacovella replied:
I didn't change this because this combination of yaml files built pretty fast. But I think it would be better for consistency (and environment caching).  So I think it is a good idea to make these args.

apayne97 replied:
that makes sense!

ijpulidos replied:
Yes, +1 for ding this in the `create_args` section instead. I wouldn't expect it to take much longer and it does help with consistency.

apayne97 commented on devtools/conda-envs/ubuntu-latest/all.yaml:
```diff
@@ -0,0 +1,111 @@
+  - pytorch_sparse
+  - scipy >=1.13.1
+  - torchdata <=0.9.0
+  - wandb
```
I think these will be easier to keep track of if they are alphabetical instead of organized by purpose, even though it is kinda nice to have them like this there isn't always a 1:1 map between the purpose and the category.

chrisiacovella replied:
This was just my attempt to put together a single yaml that would allow someone to build the environment.  I tried to keep it grouped like this to make it easier for me to update all the pins.  We should probably double check this since I went through and made pins consistent on all the other yamls.

ijpulidos replied:
While I do see the motivation to have this, my suggestion is to get rid of this file, this will probably get out of date quickly and since it's not being tested/used in any CI or workflow it's just going to become unmaintained. So far, the best one I could come up with for a dev environment is with the instructions in https://docs.drugforge.com/en/latest/installation.html#developer-installation-from-source (but this still could miss some dependencies, especially if new ones are added in a release that hasn't been deployed in `conda-forge`).

The longer term solution to this is probably moving to something like `pixi` (or similar) which would allow specifying different environments, namely one for development purposes, which could also be used in CI.

chrisiacovella replied:
Seems reasonable to just delete it.  This was mostly useful when we were having trouble with conflicts as we were updating  things.

apayne97 commented on drugforge-workflows/drugforge/workflows/spectrum_workflows/score_complex.py:
```diff
@@ -256,6 +256,7 @@ def score_complex_workflow(inputs: ScoreInputs):
                     inputs.target,
                     comp_name,
                 )
+                # chain_dock = "1"  # Standard in OpenMM output file
```
can this be deleted?

chrisiacovella replied:
You previously commented this on one of the PRs that got merged into this.  I commented it out to get pre-commit to stop complaining about the unused variable.  I guess the question here for someone who is more familiar with the scoring; why was this set and not used?  Should it be used?

apayne97 replied:
LOL sorry for the double comment

apayne97 reviewed (dismissed):
Honestly not sure how to give great feedback for this PR. but here are some minor things, I think unrelated to @kaminow 's main additions.

ijpulidos commented on .github/workflows/workflows.yaml:
```diff
@@ -32,8 +32,28 @@ jobs:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macOS-latest]
```
Pretty sure this is not case-sensitive, but just in case this ever changes. Docs suggest to use all lower case.

```suggestion
        os: [ubuntu-latest, macos-latest]
```

ijpulidos commented on devtools/conda-envs/macos-latest/workflows.yaml:
```diff
@@ -4,6 +4,6 @@ channels:
-  click
-  pydantic
-  unittest
+ click
+ pydantic
+ unittest
```
```suggestion
  - click
  - pydantic
  - unittest
```
This is probably not an issue but good for consistency/readability.

ijpulidos commented on devtools/conda-envs/ubuntu-latest/data.yaml:
```diff
@@ -2,6 +2,7 @@ name: data
 channels:
   - openeye
   - conda-forge
+  - rapidsai
```
Why do we need this channel again?

chrisiacovella replied:
I put a note about his above, replicated below: 

 Note, I had to add in the rapidsai channel for dask-cuda. the conda-forge version is really out of date and it looks like they are distributing on their own channel (which is rapidsai). Newer versions of openfe, which are required for pydantic v2, have specific requirements for dask>=2025 ... but the latest dask-cuda is 2023 and is incompatible with dask 2025.

ijpulidos commented on devtools/conda-envs/ubuntu-latest/docking.yaml:
```diff
@@ -2,11 +2,13 @@ name: docking
 channels:
   - openeye
   - conda-forge
+  - rapidsai
```
Why do we need this channel? We should _try_ to avoid extra channels, as possible.

ijpulidos commented on devtools/conda-envs/ubuntu-latest/ml.yaml:
```diff
@@ -2,11 +2,13 @@ name: ml
 channels:
   - openeye
   - conda-forge
+  - rapidsai
```
Why do we need this channel?

ijpulidos commented on devtools/conda-envs/ubuntu-latest/simulation.yaml:
```diff
@@ -2,6 +2,7 @@ name: simulation
 channels:
   - openeye
   - conda-forge
+  - rapidsai
```
Why do we need this channel?

ijpulidos commented on devtools/conda-envs/ubuntu-latest/spectrum.yaml:
```diff
@@ -2,6 +2,8 @@ name: spectrum
 channels:
   - openeye
   - conda-forge
+  - rapidsai
```
Why do we need this channel?

ijpulidos commented on devtools/conda-envs/ubuntu-latest/workflows.yaml:
```diff
@@ -4,6 +4,6 @@ channels:
-  click
-  pydantic
-  unittest
+ click
+ pydantic
+ unittest
```
```suggestion
  - click
  - pydantic
  - unittest
```

ijpulidos commented on drugforge-alchemy/drugforge/alchemy/schema/base.py:
```diff
@@ -2,7 +2,75 @@
+        val = v["val"]
+        out[key] = unit_ * val
+
+    return out
```
I don't see these being used anywhere else in the code, it seems like we don't need them. Could we remove this code?

Also, if we really want/need a `Quantity` encoder kind of object we probably want to use the `gufe` capabilities (as in https://gufe.openfree.energy/en/latest/how-tos/custom_quantities.html). Especially since `gufe` is already a dependency for `drugforge-alchemy`.

chrisiacovella replied:
Yeah, I think you are correct, the switch th GUFE quantities makes this possibly not needed.  I will remove this and see if the tests keep passing...if they fail I'll put it back.

ijpulidos commented on drugforge-alchemy/drugforge/alchemy/schema/fec.py:
```diff
@@ -1,10 +1,16 @@
+    KCalPerMolQuantity,
+    NanometerQuantity,
+    specify_quantity_units,
+)
```
Right, this is related to my previous comment in https://github.com/choderalab/drugforge/pull/2/changes#r2991069316 this module is probably where these quantity encoder objects were used, but now we are using the `gufe` ones here directly. So I guess we could get rid of the class that's pointed in the referenced object.

ijpulidos commented on drugforge-cli/drugforge/cli/tests/test_meta_cli.py:
```diff
@@ -29,7 +30,10 @@ def test_toplevel_runnable():
-            marks=pytest.mark.xfail(reason="ML package is not currently available."),
+            marks=pytest.mark.xfail(
+                condition=os.getenv("RUNNER_OS") == "macOS",
+                reason="ML package is not currently available.",
```
Just for clarity, should we note here (or somewhere) why is ML not available for macOS? I presume it's lack of packages for it, but we should probably have that documented somewhere so can be referenced/fixed in the future if anything changes.

chrisiacovella replied:
@kaminow Is there a specific package that doesn't work on macOS or extra stuff that needs installing?

ijpulidos commented on drugforge-cli/drugforge/cli/cli.py:
```diff
@@ -35,18 +35,14 @@ def cli(help="Command-line interface for drugforge"): ...
 
+    cli.add_command(ml)
+except ImportError:
+    print("ML package not available, skipping ML CLI command.")
```
We should probably use either a logger here, or the `warnings.warn` capabilities. Not sure if we have a logger setup at this point, though.

```suggestion
    warnings.warn("ML package not available, ML CLI command disabled.")
```

chrisiacovella replied:
Good catch.

ijpulidos commented on drugforge-data/drugforge/data/operators/state_expanders/expansion_tag.py:
```diff
@@ -31,8 +31,7 @@ class StateExpansionTag(BaseModel):
 
-    class Config:
-        allow_mutation = False
+    model_config = {"frozen": True}
```
Not a blocker. We probably would benefit from making this a pydantic `ConfigDict` instead of a plain `dict`. In this case the effect is the same, but just in case we want to change something in the future that takes advantage of the pydantic config class.

chrisiacovella replied:
Good catch... this one must have slipped through all the updates.

ijpulidos commented on drugforge-data/drugforge/data/schema/complex.py:
```diff
@@ -49,14 +49,21 @@ class Complex(ComplexBase):
     ligand: Ligand = Field(description="Ligand schema object")
-    ligand_chain: str = Field(None, description="Chain ID of ligand in complex")
+    # this needs to be marked as optional to allow it to be set to None in cases where this is not defined
+    ligand_chain: str | None = Field(None, description="Chain ID of ligand in complex")
```
I don't know if this is what the comment is suggesting, but wouldn't this need to be
```suggestion
    ligand_chain: Optional[str] = Field(None, description="Chain ID of ligand in complex")
```
instead?

chrisiacovella replied:
Ah, that comment is really outdated... basically pydantic v1 defaulted the value to None, while in v2 we need to be explicit (in v2 Optional doesn't imply none)....but that is ultimately done with `Field(None, ` now sooo I'm also going to just remove the comment.

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -1,16 +1,16 @@
 
 class ExperimentalCompoundData(BaseModel):
-    compound_id: str = Field(
+    compound_id: str | None = Field(
```
Not a blocker, but I think this is more "pydantonic"?
```suggestion
    compound_id: Optional[str] = Field(
```

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -1,16 +1,16 @@
     )
 
-    smiles: str = Field(
+    smiles: str | None = Field(
```
Same here, not a blocker
```suggestion
    smiles: Optional[str] = Field(
```

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -35,13 +35,25 @@ class ExperimentalCompoundData(BaseModel):
     )
 
-    date_created: date = Field(None, description="Date the molecule was created.")
+    date_created: date | None = Field(
```
Same here, not a blocker
```suggestion
    date_created: Optional[date] = Field(
```

ijpulidos commented on drugforge-data/drugforge/data/schema/identifiers.py:
```diff
@@ -93,8 +91,7 @@ class TargetIdentifiers(DataModelAbstractBase):
 
-    class Config:
-        allow_mutation = False
+    model_config = {"frozen": True, "arbitrary_types_allowed": True}
```
Not a blocker. Should we be using ConfigDict instead?
```suggestion
    model_config = ConfigDict(frozen=True, arbitrary_types_allowed=True)
```

ijpulidos commented on drugforge-data/drugforge/data/schema/legacy.py:
```diff
@@ -1,10 +1,9 @@
 class CrystalCompoundData(BaseModel):
-    class Config:
-        extra = "forbid"
+    model_config = {"extra": "forbid"}
```
Not a blocker. Should we use ConfigDict instead?
```suggestion
    model_config = ConfigDict(extra="forbid")
```

ijpulidos commented on drugforge-data/drugforge/data/tests/test_data_cli.py:
```diff
@@ -51,4 +51,35 @@ def test_cdd_to_schema(cdd_to_schema_files, tmp_path):
+    for d1, d2 in zip(json_check, json_test):
+        equalivance.append(compare_vals_in_dict(d1, d2))
+
+    assert all(equalivance)
```
Surely just a typo
```suggestion
    equivalence = []
    for d1, d2 in zip(json_check, json_test):
        equivalence.append(compare_vals_in_dict(d1, d2))

    assert all(equivalence)
```

ijpulidos replied:
If we apply the changes in the [previous comment](https://github.com/choderalab/drugforge/pull/2/changes#r2991596204) we also want to change the `compare_vals_in_dict` to just `compare_vals`. Just to keep this in mind.

ijpulidos commented on drugforge-data/drugforge/data/tests/test_data_cli.py:
```diff
@@ -51,4 +51,35 @@ def test_cdd_to_schema(cdd_to_schema_files, tmp_path):
+            for key in d1.keys():
+                if not compare_vals_in_dict(d1[key], d2[key]):
+                    return False
+            return True
```
This code is really hard to read, and I also think it's not doing what we need it to do, since it could give a false positive result when `d2` has keys that `d1` doesn't, which I think defeats the purpose. 

I believe we could write it in a more readable way as follows, recursing using the values.

```suggestion
    def compare_vals(v1, v2):
        # Handle exact equality ("pure" values)
        if v1 == v2:
            return True

        # Handle the NaN == None case
        is_v1_nan_none = v1 is None or (isinstance(v1, float) and math.isnan(v1))
        is_v2_nan_none = v2 is None or (isinstance(v2, float) and math.isnan(v2))
        
        if is_v1_nan_none and is_v2_nan_none:
            return True

        # Handle dictionaries recursively
        if isinstance(v1, dict) and isinstance(v2, dict):
            # Ensure they have the exact same keys
            if v1.keys() != v2.keys():
                return False
            # Recurse through values
            return all(compare_vals(v1[k], v2[k]) for k in v1)

        return False
```

chrisiacovella replied:
I wouldn't say that is so much easier to read, as it still has some ugly recursion going on :-P ,  but it is definitely a bit cleaner and avoids that false positive case! I like it.

ijpulidos commented on drugforge-dataviz/drugforge/dataviz/gif_viz.py:
```diff
@@ -17,8 +17,7 @@
 from drugforge.dataviz.visualizer import VisualizerBase
 from drugforge.docking.docking_data_validation import DockingResultCols
 from drugforge.simulation.simulate import SimulationResult
-from multimethod import multimethod
```
Love to see multimethod gone :)

ijpulidos commented on drugforge-docking/drugforge/docking/tests/test_scorers.py:
Love the lazy imports here. Great!

ijpulidos commented on drugforge-ml/drugforge/ml/tests/test_mlops_cli.py:
```diff
@@ -3,9 +3,17 @@
+
+try:
+    from drugforge.ml.cli_mlops import mlops as cli
+except Exception:
```
We should avoid checking for generic exceptions here. What is the actual error/exception we are trying to catch? Assuming there's a "small" set of them.

chrisiacovella replied:
I think this is fine for now because below it is just does a pytest skip with a descriptive message.  @kaminow These are from the ML tests so don't want to speak for you.

kaminow replied:
I think these tests were actually added by Hugo, and they're testing the ML ops production stuff that he set up for ASAP. for now, the ML ops stuff isn't updated to the most recent version of `mtenn` (I guess probably something I should do for a future PR/milestone), so this is just skipping the tests if importing something for `cli_mlops` fails. probably can just `xfail` or something, not sure what the appropriate thing to do from a code perspective is but it is also probably fine for now

ijpulidos commented on drugforge-ml/drugforge/ml/analysis.py:
Not a blocker. We probably want in the future to separate the CLI parts of this in their own module or in the cli.py module, something to discuss further.

ijpulidos replied:
I think we need docstrings for all these callables here, with basic usage/args and a quick description of what they are doing and how, as possible. Some of them already have docstrings but I think we need docstrings for the rest of them, those who are not obvious from the code of course. This one I think is blocking.

ijpulidos commented on drugforge-ml/drugforge/ml/asap_models.yaml:
```diff
@@ -1,3 +1,6 @@
+## Maybe want to rename these back to asapdiscovery or just get rid of them? Since they
+## were trained on the ASAP data
```
I'm okay either way, but we should make sure that we reduce the chance of there being some conflict with other models from `asapdiscovery`, since those should be using different code for it. But I'm happy with whatever @kaminow decides for this.

kaminow replied:
personally I think it probably makes sense to just leave them as asap, that way their provenance is clear. we also have the publish dates included in the names, so there shouldn't really be any risk of conflicts

ijpulidos commented on drugforge-ml/drugforge/ml/cli.py:
```diff
@@ -64,45 +62,170 @@ def train():
+    cpd_regex: str = MOONSHOT_CDD_ID_REGEX,
+    overwrite_ds_config_cache: bool = False,
+    overwrite_ds_cache: bool = False,
+):
```
I think we would like a docstring here with some description of what and how this is doing what it is doing.

ijpulidos commented on drugforge-ml/drugforge/ml/cli.py:
We want a docstring here with a summary of what and how it's doing what it's doing.

ijpulidos commented on drugforge-ml/drugforge/ml/es.py:
This module/file could benefit from having a more explicit name like `early_stopping.py`

ijpulidos commented on drugforge-ml/drugforge/ml/schema.py:
```diff
@@ -53,27 +53,29 @@ class TrainingPrediction(BaseModel):
+    model_config = {
+        "arbitrary_types_allowed": True,
+        "validate_assignment": True,
+    }
```
I think we want to homogenize here and use a `ConfigDict` instead.

ijpulidos commented on drugforge-ml/drugforge/ml/schema.py:
```diff
@@ -105,21 +107,21 @@ class TrainingPredictionTracker(BaseModel):
-    class Config:
-        # Allow things to be added to the object after initialization/validation
-        extra = Extra.allow
+    model_config = {
```
Same here, we probably would want to use `ConfigDict`

ijpulidos commented on drugforge-spectrum/drugforge/spectrum/tests/test_score.py:
```diff
@@ -66,7 +66,7 @@ def test_dock_score(protein_path):
     )
     assert scores_df["docking-score-POSIT"].values[0]
-    assert isinstance(ligand_pose, Ligand)
+    assert type(ligand_pose) is Ligand
```
Any specific reason why we want this instead of using the `isinstance`? With `isinstance` we can be safe even with subclasses, which I think should be fine for the test.

ijpulidos reviewed (changes requested):
Awesome work! I have some non-blocking comments and some suggested changes and questions. Happy to further discuss them.

chrisiacovella committed:
Addressing comments from Ivan and Alex

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
Addressing comments from Ivan and Alex

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

chrisiacovella committed:
Addressing comments from Ivan and Alex

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
Addressing comments from Ivan and Alex

chrisiacovella committed:
Addressing comments from Ivan and Alex

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
Addressing comments from Ivan and Alex

chrisiacovella committed:
reverting quantity encoder changes

chrisiacovella committed:
making precommit happy.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
making precommit happy.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
making precommit happy.

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

# Conflicts:
#	drugforge-data/drugforge/data/schema/complex.py

chrisiacovella committed:
making precommit happy.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

chrisiacovella committed:
making precommit happy.

chrisiacovella committed:
Merge remote-tracking branch 'origin/add-split-model' into add-split-model

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

kaminow committed:
Use Optional typing.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

kaminow committed:
Add docstring for ML CLI build_ds function.

chrisiacovella committed:
making precommit happy.

kaminow committed:
Fix early return for build_ds function.

chrisiacovella committed:
Adding in some doc strings and adding notes about possible sporadic failure of test_alchemy_prep_run_bad_chemistry.

kaminow committed:
Add ml.analysis docstrings.

kaminow committed:
Remove unused functions.

kaminow committed:
Add ml.cli docstrings.

kaminow committed:
Rename es -> early_stopping.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

kaminow committed:
Remove unused function.

kaminow committed:
Remove unused imports.

kaminow committed:
Renaming some missed es instances.

kaminow committed:
Match CLI args.

ijpulidos commented on .github/workflows/workflows.yaml:
```diff
@@ -32,8 +32,28 @@ jobs:
+              --file devtools/conda-envs/ubuntu-latest/dataviz.yaml
+              --file devtools/conda-envs/ubuntu-latest/spectrum.yaml
+              --file devtools/conda-envs/ubuntu-latest/ml.yaml
+          - os: macOS-latest
```
```suggestion
          - os: macos-latest
```
This doesn't seem like it mattes, but better to be consistent here. Non-blocking comment.

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -1,16 +1,16 @@
 
 class ExperimentalCompoundData(BaseModel):
-    compound_id: str = Field(
+    compound_id: Optional[str] = Field(
```
Not blocking. But I think "Type | None is preferred over Optional[Type] for Python 3.10+."  from https://stackoverflow.com/a/69440627 
```suggestion
    compound_id: str | None = Field(
```

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -1,16 +1,16 @@
     )
 
-    smiles: str = Field(
+    smiles: Optional[str] = Field(
```
Not blocking. But I think "Type | None is preferred over Optional[Type] for Python 3.10+."  from https://stackoverflow.com/a/69440627 
```suggestion
    smiles: str | None = Field(
```

ijpulidos commented on drugforge-data/drugforge/data/schema/experimental.py:
```diff
@@ -35,13 +35,25 @@ class ExperimentalCompoundData(BaseModel):
     )
 
-    date_created: date = Field(None, description="Date the molecule was created.")
+    date_created: Optional[date] = Field(
```
Not blocking. But I think "Type | None is preferred over Optional[Type] for Python 3.10+."  from https://stackoverflow.com/a/69440627 
```suggestion
    date_created: date | None = Field(
```

ijpulidos reviewed (approved):
Looks great! Awesome work. Just a couple of non-blocking comments/suggestions that are optional.

kaminow committed:
Switch macOS -> macos.

kaminow committed:
Rename models from 'drugforge*' back to 'asapdiscovery*' to denote that models were trained on ASAP data.
